qcodes/loops.py

```diff
@@ -891,11 +894,20 @@ def _run_loop(self, first_delay=0, action_indices=(),
                 if t - last_task >= self.bg_min_delay:
                     try:
                         self.bg_task()
-                    except Exception:
+                        self.last_task_failed = False
+                    except (ClosedError, ConnectionResetError) as err:
+                        # old qt process dead, we spawn a new
+                        log.error("Remote QT process "
+                                  "crashed, restarting:")
+                        log.exception(err)
+                        if isinstance(self.bg_task.__self__, QtPlot):
+                            self.bg_task.__self__._init_qt()
```

Comment:
We should probably add an else clause here that will re throw the error so that it's handled by  the usual error handler if its not a QTPlot